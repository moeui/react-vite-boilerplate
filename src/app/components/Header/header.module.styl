.header
    clear: both
    position fixed
    top: 0
    left 0
    width: 100%
    background: #fff
    box-shadow: 0px 1px 8px rgba(#000, .12)
    clear: both
    z-index: 1

    .logo
        width: 97px
        height: 32px
        background: #000
        background-size: 100% 100%
        float: left
        margin: 4px 0 0
        cursor: pointer
