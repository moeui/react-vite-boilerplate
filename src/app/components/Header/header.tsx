import * as React from 'react'
import { RouteComponentProps, withRouter } from 'react-router-dom'

import style from './header.module.styl'

interface IProps extends RouteComponentProps {
    type?: string[]
    systemName?: string
    children?: JSX.Element[] | JSX.Element
}

const Header: React.FunctionComponent<IProps> = (props: IProps): React.ReactElement => {
    const jump = (target: string): void => {
        props.history.push(target)
    }
    return (
        <header className={style.header}>
            <div className={style.logo} onClick={() => jump('/')}></div>
            {props.children}
        </header>
    )
}

export default withRouter(Header)
