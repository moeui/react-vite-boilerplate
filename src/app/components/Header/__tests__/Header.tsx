import { render } from '@testing-library/react'
import { mount, shallow } from 'enzyme'
import * as React from 'react'
import { MemoryRouter } from 'react-router-dom'

import Header from '../index'

describe('<Header />', () => {
    it('renders', () => {
        const tree = render(
            <MemoryRouter>
                <Header>
                    <div className="text" />
                </Header>
            </MemoryRouter>
        ).container.firstChild
        expect(tree).toMatchSnapshot()
    })
    it('renders', () => {
        const tree = render(
            <MemoryRouter>
                <Header />
            </MemoryRouter>
        ).container.firstChild
        expect(tree).toMatchSnapshot()
    })

    it('renders', () => {
        const tree = render(
            <MemoryRouter>
                <Header>
                    <div>test</div>
                    <div>test2</div>
                </Header>
            </MemoryRouter>
        ).container.firstChild
        expect(tree).toMatchSnapshot()
    })
    it('test click jump', () => {
        const header = mount(
            <MemoryRouter>
                <Header />
            </MemoryRouter>
        )
        header.find('div').simulate('click')
        expect(header.find('div')).toHaveLength(1)
    })
})
