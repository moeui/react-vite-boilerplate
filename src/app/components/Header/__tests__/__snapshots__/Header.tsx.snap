// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders 1`] = `
<header
  class="header"
>
  <div
    class="logo"
  />
  <div
    class="text"
  />
</header>
`;

exports[`<Header /> renders 2`] = `
<header
  class="header"
>
  <div
    class="logo"
  />
</header>
`;

exports[`<Header /> renders 3`] = `
<header
  class="header"
>
  <div
    class="logo"
  />
  <div>
    test
  </div>
  <div>
    test2
  </div>
</header>
`;
